userd/launcher.go

```diff
@@ -93,3 +102,78 @@ func (s *Launcher) OpenURL(addr string) *dbus.Error {
 
 	return nil
 }
+
+func fdToFilename(fd int) (string, error) {
+	flags, err := sys.FcntlGetFl(fd)
+	if err != nil {
+		return "", err
+	}
+	// File descriptors opened with O_PATH do not imply access to
+	// the file in question.
+	if flags&sys.O_PATH != 0 {
+		return "", fmt.Errorf("bad file descriptor")
```

Comment:
The error message could be confusing, how about `cannot use file descriptors opened using O_PATH`